android/app/src/main/java/com/sample/ComponentActivity.kt

```diff
@@ -0,0 +1,116 @@
+package com.sample
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.KeyEvent
+import androidx.appcompat.app.AppCompatActivity
+import com.facebook.react.PackageList
+import com.facebook.react.ReactInstanceManager
+import com.facebook.react.ReactRootView
+import com.facebook.react.common.LifecycleState
+import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler
+import com.facebook.soloader.SoLoader
+
+class ComponentActivity : AppCompatActivity(), DefaultHardwareBackBtnHandler {
+    private lateinit var reactRootView: ReactRootView
+    private lateinit var reactInstanceManager: ReactInstanceManager
+
+    companion object {
+        private const val COMPONENT_NAME = "extra:componentName";
+
+        fun newIntent(activity: Activity, componentName: String): Intent {
+            return Intent(activity, ComponentActivity::class.java).apply {
+                putExtra(COMPONENT_NAME, componentName)
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        SoLoader.init(this, false)
+
+        val componentName = intent.getStringExtra(COMPONENT_NAME)
+
+        reactRootView = ReactRootView(this)
+        setContentView(reactRootView)
+
+        val reactInstanceManagerBuilder = ReactInstanceManager.builder()
+            .setInitialLifecycleState(LifecycleState.BEFORE_RESUME)
+            .addPackages(PackageList(application).packages)
+            .setApplication(application)
+            .setCurrentActivity(this)
+
+        val bundleName = findBundleFile()
+        if (bundleName == null) {
+            reactInstanceManagerBuilder
+                .setUseDeveloperSupport(BuildConfig.DEBUG)
+                .setJSMainModulePath("index")
```

Comment:
A note for myself: check if iOS parameterises the module path name when fetching the bundle from dev server. cc @tido64

Reply:
Fallback to the one provided by default in the `ReactNativeHost`